src/veh_interact.cpp

```diff
@@ -316,30 +315,19 @@ task_reason veh_interact::cant_do (char mode)
         } );
         has_tools = true; // checked later
         break;
-    case 'f': // refill mode
-        if (!ptanks.empty()) {
-            std::vector<vehicle_part *>::iterator iter;
-            for (iter = ptanks.begin(); iter != ptanks.end(); ) {
-                if ((*iter)->hp > 0 &&
-                    g->refill_vehicle_part(*veh, *iter, true)) {
-                    pass_checks = true;
-                    iter++;
-                } else {
-                    iter = ptanks.erase(iter);
-                }
-            }
-            if(pass_checks) {
-                return CAN_DO;
-            }
-        }
 
-        // if refillable parts exist but can't be refilled for some reason.
-        if (has_ptank) {
-            return CANT_REFILL;
-        }
+    case 'f': {
+        // first find all possible ammo
+        auto opts = g->u.items_with( []( const item &e ) { return e.is_ammo(); } );
```

Comment:
> It does not enable it, it's already there.

But it is entirely unused. If it it works, completely and without bugs, it's purely by chance as opposed to conscious design. Given sufficient time such support will only bitrot. Sure we could write test cases to prevent this, but how is that not YAGNI - we don't use it now nor is there a clear need to support it.

This versus a precondition that allows the other code to be simpler. You can always extend from that if you do decide you need it.

> Container items don't require their content to be ammo.

We probably should work on container mechanics to allow restriction on their contents. At that time they can become a direct replacement for vehicle tanks.